controllers/dashboard/users.controller.js

```diff
@@ -53,46 +99,19 @@ function AccountsController(options) {
       else if (!UserModel.validatePassword(password, user.password))
         errors.push('Mot de passe invalide');
       else if (!user.emailConfirmation) {
-        return EmailService
-          .sendEmailConfirmation(user, res._dashboarduri)
-          .then(function(result) {
-            if (result.responseStatus.accepted.indexOf(user.email) == -1) {
-              return FlashHelper.addError(req.session, 'Votre adresse E-Mail a été rejetée par le serveur de messagerie lors de l\'envoi du courriel de confirmation', function (err) {
-                if (err)
-                  return next({code: 500});
-                return res.redirect('/login').end();
-              });
-            }
-            FlashHelper.addError(req.session, {
-              title: 'Votre adresse E-Mail n\'a pas été confirmée',
-              icon: 'send',
-              messages: [
-                'Un nouveau lien de confirmation vous a été envoyé',
-                'Date d\'expiration du lien : ' + result.emailToken.accessTokenExpiresOn.toLocaleString(),
-                'Les liens envoyés précédemment deviennent inactifs'
-              ]
-            }, function(err) {
-              if (err)
-                return next({code: 500});
-              logger.info(JSON.stringify(result));
-              return res.redirect('/login');
-            });
-          })
-          .catch(function(err) {
-            if (err.code === 503) {
-              return FlashHelper.addError(req.session, {
-                title: 'Une erreur est survenue',
-                icon: 'mail',
-                messages: err.messages
-              }, function (err) {
-                if (err)
-                  return next({code: 500});
-                return res.redirect('/login');
-              });
-            }
-            return next({code: 500});
-          });
+        errors.push('L\'adresse email n\'a pas été confirmée, veuillez consultez vos emails');
+
+        sendEmailConfirmation(user, req, res, function(err) {
+          if (err) {
+            return next(err);
```

Comment:
We should handle error here because `err` is defined when something really goes wrong on server (inserting data in database).
The other errors are simply notified, but `err` keep undefined